### DISCUSSION TITLE
Dynamic values in OpenAPI Example requests

### DISCUSSION THREAD

**Discussion by noelmccrory** [specmatic/specmatic#1136] [Ask For Help]
One of the endpoints on my API accepts a `startDate` field in the request body. The API also has validation to ensure that the startDate is no more than 5 days in the past, and no more than 5 days in the future. In the example for this endpoint in the OpenAPI Spec I can specify today's date and the Contract Tests will pass successfully. However, in 6 days the Contract Test for this endpoint will fail because a 400 will be returned.

Is there any way to provide dynamic values in examples in OpenAPI Specs? I couldn't see anything in the Specmatic documentation or previous issues/discussions on this.

The only alternative I can think of at this point would be to mock out the Clock instance on the service, so that today is always a fixed date.

Thanks.

**Comment by yogeshnikam671** [comment]
Hey @noelmccrory, thanks for starting this thread. This is a very good scenario that you have brought up. 
We are indeed working on it. Will keep you posted.

**Comment by MikeJonesRVU** [comment]
Hi both. I have the same requirementin my own project. Is there any update on this functionality?

**Reply by mik047** [reply]
Hello @MikeJonesRVU , thanks for reaching out. 

There are no updates on this yet. Will keep you posted.

In the meantime, we are happy to accept a PR on this enhancement.
